Readme.md

```diff
@@ -19,6 +19,99 @@ Settings can be specified in a TOML configuration file, the path to which can it
 
 	CONFIG_FILE=config.toml uvicorn --app-dir scripts archive_api:app
 
+### Local testing in Python virtual environment
+
+```bash
+$ python3.9 -m venv venv
+$ . venv/bin/activate
+(venv)
+$ python -m pip install -r requirements.txt 
+$ CONFIG_FILE=config.toml uvicorn --app-dir=./scripts/ --port 8888 archive_api:app
+
+Loading configuration from config.toml
+2023-09-29 13:40:23,040:utility_api.py:INFO:Basic logging is configured at INFO
+2023-09-29 13:40:23,040:archive_api.py:INFO:{'config_file': None, 'log_level': 'INFO', 'log_format': '%(asctime)s:%(filename)s:%(levelname)s:%(message)s', 'db_type': 'mock', 'db_host': None, 'db_port': None, 'db_name': None, 'db_username': None, 'db_log_frequency': 100, 'db_aws_secret_name': None, 'db_aws_region': 'us-west-2', 'store_type': 'S3', 'store_primary_bucket': 'hopskotch-archive', 'store_backup_bucket': 'hopskotch-archive-backup', 'store_endpoint_url': None, 'store_region_name': None, 'store_log_every': 100, 'read_only': False, 'hop_auth_api_root': None}
+2023-09-29 13:40:23,040:database_api.py:INFO:Mock Database configured
+INFO:     Started server process [49466]
+INFO:     Waiting for application startup.
+2023-09-29 13:40:23,045:archive_api.py:INFO:Connecting to archive
+2023-09-29 13:40:25,363:archive_api.py:INFO:Startup complete
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://127.0.0.1:8888 (Press CTRL+C to quit)
+```
+
+### Local testing in Docker Compose
+
+The Docker Compose file defines a local deployment of the following components:
+
+- archive-api: the archive API webserver (http://localhost:8000)
+- archive-db: a PostgreSQL database for the archive metadata
+- archive-ingest: the script that consumes Hopskotch messages and stores their metadata in the archive db and their payload in the object store
+- object-store: an instance of MinIO for S3-compatible object storage for the archive data storage (http://localhost:9001/browser)
+
+#### Build the archive-ingest image
+
+Clone the archive-ingest repo and build the Docker image:
+
+```bash
+git clone git@github.com:scimma/archive-ingest.git
+cd archive-ingest
+docker build . -t scimma-archive-ingest:dev
+```
+
+#### Obtain Hopskotch credentials
+
+By default you will need to provide Hopskotch credentials `HOP_USERNAME` and `HOP_PASSWORD` in a `.env` file as illustrated in the `env.default` file. Any environment variables defined in `.env` will override the values specified in `env.default`. 
```

Comment:
This is one of the subheading under the "Local testing in Docker Compose" section, and it is a step in preparing the local Compose environment. The `.env` file is used to populate the environment within the relevant containers, so that the hop clients therein have the credentials they need.

Reply:
It would be cleaner to use the standard mechanism for handling hop credentials and not clutter the top level directory with more dot files, particularly if they contain credentials that people should be handling with some care.